hive/runners/multi_agent_loop.py

```diff
@@ -130,19 +135,23 @@ def run_end_step(self, episode_metrics, done=True):
     def run_episode(self):
         """Run a single episode of the environment."""
         episode_metrics = self.create_episode_metrics()
-        done = False
+
         observation, turn = self._environment.reset()
         self._transition_info.reset()
         steps = 0
+        terminated, truncated = False, False
+
         # Run the loop until the episode ends or times out
-        while not done and steps < self._max_steps_per_episode:
-            done, observation, turn = self.run_one_step(
+        while not (terminated or truncated) and steps < self._max_steps_per_episode:
+            terminated, truncated, observation, turn = self.run_one_step(
                 observation, turn, episode_metrics
             )
             steps += 1
+            if steps == self._max_steps_per_episode:
+                truncated = True
```

Comment:
again, I think `truncated=not terminated` here

Reply:
done.